Dixit_Logic/Classes/GameState.cs

```diff
@@ -71,15 +71,27 @@ public class GameState : IGameState
         public GameState()
         {          
             _storyTeller = null;
-            //_boardDeck = new IDeck();
+            _boardDeck = new Deck();
             _storyText = "";
             _guesses = new Dictionary<IPlayer, ICard>();
-            _hands = new Dictionary<IPlayer, IDeck>();
-            //_mainDeck = new IDeck();
-            _baseCards = new List<ICard>().AsReadOnly();
+            _hands = new Dictionary<IPlayer, IDeck>();        
             _players = new List<IPlayer>();            
             _points = new Dictionary<IPlayer, int>();
             _roundStatus = PhaseStatus.BeforeStart; // set to the game's round starting status
+
+            
+            List<ICard> baseCardList = new List<ICard>();
+
+            const int numberOfCards = 84;
+            //add cards to main deck
```

Comment:
@matyimat How do you know the number of cards? This should not be hardcoded, as this part of the code is not responsible for the data. Data layer provides an interface for getting back the list of cards.